added support for colons in variable names in ExecComp

fzahle opened the pull request:
### Summary

ExecComp does not support colons in variable names, but the rest of the framework does. This commit ports this functionality from OpenMDAO1.

### Related Issues

- Resolves #

### Backwards incompatibilities

None

### New Dependencies

None

JustinSGray commented:
can you add a test that has an exec-comp with 2 separate equations, and the same input variable in both of them? What about an expression like `foo:bar = tree:leaf_count + tree:leaf_count`? Will the duplication of the input be handled correctly in either case?

fzahle commented:
colon variables of course have the drawback that array sizes can't be initialised in the instantiation since they aren't valid python names:

    om.ExecComp('foo:bar=x:bar+1.', x:bar=2.0)

but can of course be set through a dictionary. I just converted some code from OpenMDAO1 and came across this difference. Also there's a few numpy functions that aren't supported that I've added. I'll push that to another PR.
